### PR TITLE
docs(docs): nx cloud e2e encryption configuration documentation added

### PR DESCRIPTION
**Pull request by ZackDeRose**
Adding in documentation for e2e encryption for the Nx Cloud task runner.

## Related Issue(s)

Fixes #

### CODE REVIEW

**Comment by ZackDeRose** [comment]
@vsavkin, wanted to confirm with you that this information is public and not expected to change before merging!

File: docs/angular/guides/configuration.md [hunk 1]
```diff
@@ -358,3 +359,7 @@ The syntax is the same as a [`.gitignore` file](https://git-scm.com/book/en/v2/G
 
 1. Changes to that file will not be taken into account in the `affected` calculations.
 2. Even if the file is outside an app or library, `nx workspace-lint` will not warn about it.
+
+## End-to-end Cloud Encryption
```
Comment (isaacplmann) [inline docs/angular/guides/configuration.md:363]: This duplicates a lot of the text in the section above.  I would remove this section and add a not to the command line option saying "You may also provide this value as an environment variable with the key: `NX_CLOUD_ENCRYPTION_KEY`"

**Review by isaacplmann — CHANGES_REQUESTED** [review]

File: docs/angular/guides/configuration.md (hunk 1 above)
Comment (ZackDeRose) [reply docs/angular/guides/configuration.md]: Moved - good call

File: docs/angular/guides/configuration.md [hunk 2]
```diff
@@ -265,6 +265,7 @@ Tasks runners can accept different options. The following are the options suppor
 - `captureStderr` defines whether the cache will capture stderr or just stdout
 - `skipNxCache` defines whether the Nx Cache should be skipped. Defaults to `false`
 - `cacheDirectory` defines where the local cache is stored, which is `node_modules/.cache/nx` by default.
+- `encryptionKey` (when using `"@nrwl/nx-cloud"` only) defines an encryption key to support end-to-end encryption of your cloud cache. See also [End-to-end Cloud Encryption](#end-to-end-cloud-encryption) for providing this key via an environment variable. You may also provide an environment variable with the key `NX_CLOUD_ENCRYPTION_KEY` that contains an encryption key as its value. The Nx Cloud task runner will normalize the key length, so any length of key is acceptable.
```
Comment (isaacplmann) [inline docs/angular/guides/configuration.md:268]:
```suggestion
- `encryptionKey` (when using `"@nrwl/nx-cloud"` only) defines an encryption key to support end-to-end encryption of your cloud cache. You may also provide an environment variable with the key `NX_CLOUD_ENCRYPTION_KEY` that contains an encryption key as its value. The Nx Cloud task runner will normalize the key length, so any length of key is acceptable.
```

File: docs/react/guides/configuration.md [hunk 3]
```diff
@@ -288,6 +288,7 @@ Tasks runners can accept different options. The following are the options suppor
 - `captureStderr` defines whether the cache will capture stderr or just stdout
 - `skipNxCache` defines whether the Nx Cache should be skipped. Defaults to `false`
 - `cacheDirectory` defines where the local cache is stored, which is `node_modules/.cache/nx` by default.
+- `encryptionKey` (when using `"@nrwl/nx-cloud"` only) defines an encryption key to support end-to-end encryption of your cloud cache. See also [End-to-end Cloud Encryption](#end-to-end-cloud-encryption) for providing this key via an environment variable. You may also provide an environment variable with the key `NX_CLOUD_ENCRYPTION_KEY` that contains an encryption key as its value. The Nx Cloud task runner will normalize the key length, so any length of key is acceptable.
```
Comment (isaacplmann) [inline docs/react/guides/configuration.md:291]:
```suggestion
- `encryptionKey` (when using `"@nrwl/nx-cloud"` only) defines an encryption key to support end-to-end encryption of your cloud cache. You may also provide an environment variable with the key `NX_CLOUD_ENCRYPTION_KEY` that contains an encryption key as its value. The Nx Cloud task runner will normalize the key length, so any length of key is acceptable.
```

**Review by isaacplmann — CHANGES_REQUESTED** [review]

**Comment by isaacplmann** [comment]
Some leftover text that doesn't make sense after removing the second section.

File: docs/angular/guides/configuration.md [hunk 4]
```diff
@@ -265,6 +265,7 @@ Tasks runners can accept different options. The following are the options suppor
 - `captureStderr` defines whether the cache will capture stderr or just stdout
 - `skipNxCache` defines whether the Nx Cache should be skipped. Defaults to `false`
 - `cacheDirectory` defines where the local cache is stored, which is `node_modules/.cache/nx` by default.
+- `encryptionKey` (when using `"@nrwl/nx-cloud"` only) defines an encryption key to support end-to-end encryption of your cloud cache. See also [End-to-end Cloud Encryption](#end-to-end-cloud-encryption) for providing this key via an environment variable. The Nx Cloud task runner will normalize the key length, so any length of key is acceptable.
```
Comment (isaacplmann) [inline docs/angular/guides/configuration.md:268]:
```suggestion
- `encryptionKey` (when using `"@nrwl/nx-cloud"` only) defines an encryption key to support end-to-end encryption of your cloud cache. You may also provide an environment variable with the key `NX_CLOUD_ENCRYPTION_KEY` that contains an encryption key as its value. The Nx Cloud task runner will normalize the key length, so any length of key is acceptable.
```

**Review by isaacplmann — CHANGES_REQUESTED** [review]

**Review by isaacplmann — APPROVED** [review]

**Review by bcabanes — APPROVED** [review]

**Pull request merged**